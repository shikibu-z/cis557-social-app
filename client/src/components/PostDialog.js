import React from 'react';
import '../stylesheets/PostDialog.css';

function PostDialog(props) {
  const { handleClose, content } = props;
  return (
    <div className="popup-box">
      <div className="box">
        <button type="button" className="close-icon" onClick={handleClose} onKeyDown={handleClose}>x</button>
        {content}
      </div>
    </div>
  );
}

export default PostDialog;
