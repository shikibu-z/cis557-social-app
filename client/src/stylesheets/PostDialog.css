.popup-box {
  position: fixed;
  background: #00000050;
  width: 100%;
  height: 100vh;
  top: 0;
  left: 0;
}
 
.box {
  position: relative;
  width: 70%;
  margin: 0 auto;
  height: auto;
  max-height: 70vh;
  margin-top: calc(100vh - 85vh - 20px);
  background: #fff;
  border-radius: 4px;
  padding: 20px;
  border: 1px solid #999;
  overflow: auto;
}
 
.close-icon {
  content: 'x';
  cursor: pointer;
  position: fixed;
  right: calc(15% - 30px);
  top: calc(100vh - 85vh - 33px);
  background: #ededed;
  width: 25px;
  height: 25px;
  border-radius: 50%;
  line-height: 20px;
  text-align: center;
  border: 1px solid #999;
  font-size: 20px;
}

#submitPostForm {
  margin: 2%;
}

#postPopupHeader {
  text-align: left;
  font-size: 20pt;
  font-weight: 400;
  border-bottom: 1px solid black;
  margin-bottom: 2%;
}

#inputPostTitle {
  width: 100%;
  box-shadow: none;
  border-style: none;
  border: 1px solid black;
  padding: 1% 1% 1% 1%;
  margin-bottom: 2%;
}

#postTypeContainer {
  display: flex;
  width: 15%;
  flex-direction: row;
  justify-items:left;
  justify-content: space-between;
}

#inputPostContent {
  width: 100%;
  height: 100%;
  box-shadow: none;
  border-style: none;
  border: 1px solid black;
  padding: 1% 1% 1% 1%;
  margin-bottom: 2%;
}

#postAttatchments {
  margin-top: 1%;
  margin-bottom: 1%;
}