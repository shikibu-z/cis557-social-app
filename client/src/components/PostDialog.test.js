/*eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import PostDialog from './PostDialog';

test('Comment Detail matches snapshot', () => {
  const component = renderer.create(<PostDialog />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
