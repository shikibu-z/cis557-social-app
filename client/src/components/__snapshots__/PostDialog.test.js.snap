// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment Detail matches snapshot 1`] = `
<div
  className="popup-box"
>
  <div
    className="box"
  >
    <button
      className="close-icon"
      type="button"
    >
      x
    </button>
  </div>
</div>
`;
